doc/api/esm.md

```diff
@@ -1062,15 +1062,17 @@ condition list **must** be passed through to the `defaultResolve` function.
 
 ```js
 /**
- * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
+ * @param {!string} specifier
+ * @param {!{
```

Comment:
It's a bit unfortunate that this breaks syntax highlighting. Also - is there a risk that adding things like `{!{` to our docs will be more confusing than helpful to some users?

Reply:
Yeah, GitHub's parser is borked. Peeps can change the parser when sharing fenced JSDoc code snippets. I'm using `ruby` below. Regarding the exclamation point, my personal preference is to use parenthesis regardless of whether or not there are multiple possible types. So, for example, instead of ...

```ruby
/**
 * @param {!string} specifier
 * @param {!{
 *   parentURL:!(URL|string|undefined),
 *   conditions:!Array<!string>,
 * }} context
 * @param {!Function} defaultResolve
 * @returns {!{
 *   url:!string,
 * }}
 */
```
... I prefer ...
```ruby
/**
 * @param {!(string)} specifier
 * @param {!({
 *   parentURL:!(URL|string|undefined),
 *   conditions:!(Array<!(string)>),
 * })} context
 * @param {!(Function)} defaultResolve
 * @returns {!({
 *   url:!(string),
 * })}
 */
```

It seems to more clearly communicate type selection excluding null.

GitHub's parsing problem doesn't seem to come up very often because, in reality, one would be referencing types like `ResolveHookReturnObject` as opposed to these anonymous one-off objects.

Reply:
I've opted to use the minimal amount of exclamation points necessary to pass typechecking. Additionally, I've placed the only necessary exclamation point at the end of the selection to prevent ambiguity. Win, win. :)